### PR TITLE
Upgrade Snappymail to 2.21 and merge the webmail containers

### LINKED ISSUE #948: Autotest for Rainloop admin console
In relation to #947, we should create a small script in the test suite which checks if the Rainloop admin interface is indeed disabled by default. Just to be safe for any future changes in config parsing of Rainloop.

A simple implementation would be testing the HTTP return code, however Rainloop still returns 200, OK even when the page is prompting "Access Denied". See RainLoop/rainloop-webmail#1838. If that gets not solved, we might resort to parsing the result page. THat would be more error prone if they ever choose to change the layout of the error page.

### LINKED ISSUE #2250: Rainloop: Missing support for '*_ADDRESS' env vars
### Environment
`docker-compose`

### Versions
`1.9`

## Description
Rainloop's `start.py` calls `system.resolve_address()` instead of `system.get_host_address_from_environment()`, thus name resolution breaks when using `*_ADDRESS` env vars.

As a workaround one can set `HOST_FRONT` and `HOST_IMAP` manually on the `rainloop` container to get the same behaviour, but it would be great to use the existing `mailu.env` for `rainloop` too.

## Expected behaviour
`rainloop` honors `FRONT_ADDRESS` and `IMAP_ADDRESS`, when set in `mailu.env`.

### LINKED ISSUE #2466: add aspell-en,de
please add aspell-en & aspell-de packages into roundcube docker image, because spell checker is not working

Mailu version 1.9. environment docker.

PHP Warning:  pspell_new(): PSPELL couldn't open the dictionary. reason: No word lists can be found for the language &quot;en&quot;. in /var/www/html/program/lib/Roundcube/spellchecker/pspell.php on line 76, referer: https://my-domain/webmail/?_task=mail&_action=compose&_id=225630545633d558dc2f1d

Thank you.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

Upgrade Snappymail to 2.21 and merge the webmail containers. This will make the CI faster and should simplify things going forward (hardening but also allow running more than one webmail at the time, ...).

- enable APCu
- add new test to ensure we redirect to SSO and have disabled the admin panel
- add all the packaged dictionaries for spell checking
- harden the configuration of the webmails a bit (more to come in a separate PR)
- turn off deprecation warnings (php8.1 is too new)
- turn off error reporting (log them instead)
- return HTTP302 when we should
- gpg-verify the signature of the webmails we ship
- upgrade to snappymail 2.21, switch to the new json config format
- use socrates as it's meant to so that helm users can do their thing
- run the HTTPd and PHP as different users
- redirect the PHP errors to stderr

## Related issue(s)
- closes #2466
- closes #948
- closes #2250

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors retry

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors retry

**Comment by nextgens** [comment]
bors retry

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors retry

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by Diman0** [comment]
bors try

**Review by Diman0 — CHANGES_REQUESTED** [review]
See in-line comment.

All changes look fine to me. I still have to test it, before I approve it.

File: webmails/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  ; chown -R root:root /var/www/snappymail \
+  ; chown -R mailu:mailu /var/www/snappymail/data \
+  ; chown -R root:root /var/www/roundcube/ \
+  ; chown -R mailu:mailu /var/www/roundcube/temp /var/www/roundcube/logs \
+  ; chmod -R a+rX /var/www/roundcube
+
+# common
+COPY start.py /
+COPY php.ini /defaults/
+COPY php-webmail.conf /etc/php81/php-fpm.d/php-webmail.conf
```
[the remark this answers is not in the document]
Comment (nextgens) [reply webmails/Dockerfile]: done in 28d720bb

**Comment by Diman0** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by Diman0** [comment]
bors try

**Review by Diman0 — APPROVED** [review]
LGTM

**Pull request merged**